### DISCUSSION TITLE
Diference YAHFA and SandHook ?

### DISCUSSION THREAD

**Discussion by Tatuatul** [ElderDrivers/EdXposed#726] [Q&A]
what is the difference between the two?
why does it appear first YAHFA  and after SandHook ?
is there a difference between them?
who is YAHFA  and who is SandHook? thank's.

**Comment by kotori2** [comment]
> what is the difference between the two?

One spells "Y-A-H-F-A" and the other spells "S-a-n-d-H-o-o-k"

> why does it appear first YAHFA and after SandHook ?

Because YAHFA appear first then SandHook

> is there a difference between them?

Yes there are. Please refer to their technical documents. 

> who is YAHFA and who is SandHook?

It's not CyberPunk 2077. They are projects not human.

**Reply by Mekanik97** [reply]
You had one job... 😑

**Comment by Ubuntuify** [comment]
> What is the difference between the two?

They are different implementations of Hook Frameworks to Android Runtime **(ART)**. They are developed by entirely different people and there are probably little similarities to their code. The only similarity (to my knowledge) between them is their function.

> Why does it appear first YAHFA and after Sandhook? **(Why does YAHFA appear first compared to Sandhook)**

This question no longer applies as both implementations are now both in the same module. If you install it, it will default to YAHFA, you can change this in EdXposed Manager's settings.

**Please note:** EdXposed defaults to YAHFA due to it's age, stability, and wider architecture implementation like x86 support.

> Is there a difference between them?

Yes, there are differences between them. Please refer to their GitHub repositories for more information.

[Link to YAHFA repository.](https://github.com/PAGalaxyLab/YAHFA)
[Link to Sandhook repository.](https://github.com/ganyao114/SandHook)

> Who **(what)** is YAHFA and who **(what)** is Sandhook?

[Yet Another Hook Framework for ART](https://github.com/PAGalaxyLab/YAHFA) **(YAHFA)** and [Sandhook](https://github.com/ganyao114/SandHook) are hook frameworks for Android ART. 

For choosing a variant, you should default to using YAHFA as it is an older framework and therefore has more stability. ***YAHFA is 2 years older than Sandhook.***

You can use Sandhook and see if it's stable enough for you. Please note that Sandhook isn't tested on ARM32, so if you have problems on ARM32, it might be more stable on ARM64. Sandhook also does not contain any support for any x86 and x86-64.

**New edit.** You should now default to YAHFA now, instead of Sandhook as it has now become more heavy on your system then YAHFA.

**Reply by Tatuatul** [reply]
Thank you very much, a nice and detailed answer.
